tests/sentry/rules/processing/test_delayed_processing.py

```diff
@@ -1,53 +1,292 @@
+        event5 = self.create_event(self.project_two, self.now, "group-5", self.environment.name)
+        self.create_event(self.project_two, self.now, "group-5", self.environment.name)
+        self.create_event(self.project_two, self.now, "group-5", self.environment.name)
+        group5 = event5.group
+        assert group5
+        assert self.group1
+
+        self.mock_buffer.get_hash.return_value = [
+            {
+                f"{self.rule1.id}:{self.group1.id}": {self.event1.event_id},
+                f"{rule5.id}:{group5.id}": {event5.event_id},
+            },
+        ]
+
+        rules = apply_delayed(self.project, self.mock_buffer)
+        assert self.rule1 in rules
+        assert rule5 in rules
+
+    def test_apply_delayed_same_condition_diff_interval(self):
+        """
+        Test that two rules with the same condition and value but a different interval are both scheduled to fire
+        """
+        diff_interval_rule = self.create_project_rule(
+            project=self.project,
+            condition_match=[self.event_frequency_condition3],
+            environment_id=self.environment.id,
+        )
+        event5 = self.create_event(self.project.id, self.now, "group-5", self.environment.name)
+        self.create_event(self.project.id, self.now, "group-5", self.environment.name)
+        group5 = event5.group
+        assert group5
+        assert self.group1
+
+        self.mock_buffer.get_hash.return_value = [
+            {
+                f"{self.rule1.id}:{self.group1.id}": {self.event1.event_id},
+                f"{diff_interval_rule.id}:{group5.id}": {event5.event_id},
+            },
+        ]
+
+        rules = apply_delayed(self.project, self.mock_buffer)
+        assert self.rule1 in rules
+        assert diff_interval_rule in rules
+
+    def test_apply_delayed_same_condition_diff_env(self):
+        """
+        Test that two rules with the same condition, value, and interval but different environment are both scheduled to fire
+        """
+        environment3 = self.create_environment(project=self.project)
+        diff_env_rule = self.create_project_rule(
+            project=self.project,
+            condition_match=[self.event_frequency_condition],
+            environment_id=environment3.id,
+        )
+        event5 = self.create_event(self.project.id, self.now, "group-5", environment3.name)
+        self.create_event(self.project.id, self.now, "group-5", environment3.name)
+        group5 = event5.group
+        assert group5
+        assert self.group1
+
+        self.mock_buffer.get_hash.return_value = [
+            {
+                f"{self.rule1.id}:{self.group1.id}": {self.event1.event_id},
+                f"{diff_env_rule.id}:{group5.id}": {event5.event_id},
+            },
+        ]
+
+        rules = apply_delayed(self.project, self.mock_buffer)
+        assert self.rule1 in rules
+        assert diff_env_rule in rules
+
+    def test_apply_delayed_two_rules_one_fires(self):
```

Comment:
i really like how you have each of these these structured 🎉  it's really clear the cases you're testing for and the intended result.